### PR TITLE
Log payload from token passed to `authorize`

### PR DESCRIPTION
**Pull request by pbugni**
Fix for issue #11 , logging info in payload of authorize token after launch.

By default, a number of attributes found in the auth token will be stripped out before sending to the log server, to keep it clean.  Set `AUTH_TOKEN_LOG_FILTER` to a reduced set, if desired.

### CODE REVIEW

**Comment by pbugni** [comment]
@mcjustin getting closer here, need to decode, clean and parse the content...
BUT, I also think much of this is noise and isn't useful in the logs on every launch->authorize request.  Thinking I could strip out at least the following keys (or we can white list the ones we want, or keep 'em all as you see fit):

```
- access_token
- token_type
- expires_in
- scope
- id_token
- location
- loginDepartment
- need_patient_banner
- smart_style_url
```
i.e. we're just keeping from the sample, the following:
```
- state
- CSN
- DOB
- ENCDATE
- EPICUSERID
- EPTID
- FHIRPATID
- FNAME
- LNAME
- MNAME
- SYSLOGIN
- USERPROVIDERFHIRID
- __epic.dstu2.patient
- encounter
- patient
```

Here's an example of the full list of what we received from your latest attempt yesterday:
```
{
  "access_token": "eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJ1cm46ZXBpYzphcHBvcmNoYXJkLmN1cnByb2QiLCJjbGllbnRfaWQiOiJjOTE2ODg5Zi00ZTMzLTRkZmEtOTgwZC05NjZiYTQ5MzE1ZjMiLCJlcGljLmVjaSI6InVybjplcGljOkN1cnJlbnQtQXBwLU9yY2hhcmQtUHJvZHVjdGlvbiIsImVwaWMubWV0YWRhdGEiOiJ1Z09vSzh5U3F3MXdTZE52TXlGNUhuMFp3QTNlVWxnb3NMaXZiUmhNNWRJSGJOLUtjV2hsaTNhTW5rSVFGcGtzMVBFeEo3RzRqZTZ2Znc4TEM3Ui1PUlVMWXNOVzFDanp1dEpWaUFVc3pmeFU5U3lyeXRWSUtZbnZkMDIyNXhJeiIsImVwaWMudG9rZW50eXBlIjoiYWNjZXNzIiwiZXhwIjoxNjkzNDQ2NjMxLCJpYXQiOjE2OTM0NDMwMzEsImlzcyI6InVybjplcGljOmFwcG9yY2hhcmQuY3VycHJvZCIsImp0aSI6ImVmMDhlMGQ2LTE5ZTUtNDY0MC1hYzFmLTJkMDNlMTQwNjgxNSIsIm5iZiI6MTY5MzQ0MzAzMSwic3ViIjoiZUppUGNNNktpcTRtVGtBRmpQNWp5VlEzIn0.HX9EH-byxU64kGhlMtEytoCFATA87DhoJKLyWi9IYayU7BGpypPqP0X35fosBFrm6cRyrceem25EWGQYhe5xjMYrmrxzg2sjJesnId18B4IssF_qdXFLNxRJLExX6idgNiQPjcSMBJuTvbDzd7kCMjYaYNl3Uhpmwu1sBhfc5JJG4x-yoSiFurxjy_lg7mBVrqtv-PS-OjU53EFNr3ao2IuUn-HHAAMFdn55fQUp4XvL9dQ35ee6cmK-qm5JGOkoi_ZrjDXvtMKtXTvFkjqXMg0jNp0njIP28cPErZ8Xaf6Iq_DLXxfMku2sBtzOCfjBpG3x2plIYGhixiG-hte9-w",
  "token_type": "Bearer",
  "expires_in": 3600,
  "scope": "user/Binary.read user/Condition.read user/DiagnosticReport.read user/DocumentReference.read user/Encounter.read user/Medication.read user/MedicationRequest.read user/Observation.read user/Patient.read user/Practitioner.read user/PractitionerRole.read user/Procedure.read user/Questionnaire.read user/QuestionnaireResponse.Read fhirUser launch launch/encounter launch/patient openid",
  "state": "zO1Yh7J4f7lZuFvBNMei9bD9pHmjd5",
  "CSN": "39474",
  "DOB": "1%2F15%2F1987",
  "ENCDATE": "2%2F15%2F2023",
  "EPICUSERID": "++UWMED",
  "EPTID": "+++++Z4572",
  "FHIRPATID": "eJzlzKe3KPzAV5TtkxmNivQ3",
  "FNAME": "Allison",
  "LNAME": "Mychart",
  "MNAME": "",
  "SYSLOGIN": "UWMED",
  "USERPROVIDERFHIRID": "%USERPROVIDERFHIRID%",
  "__epic.dstu2.patient": "TJgJ7wgwgI32QvHx3YBhDm4egjwdyhmAhfdtE7bQsPYYB",
  "encounter": "eJHri.Y9EWJi2Lq1Yrzo5TQ3",
  "id_token": "eyJhbGciOiJSUzI1NiIsImtpZCI6IjFvcnd4eC8reFEyRVBxaUR2OHlZK0I3SGpUajRVd2QzbVRSOFNBR1hUZlE9IiwidHlwIjoiSldUIn0.eyJhdWQiOiJjOTE2ODg5Zi00ZTMzLTRkZmEtOTgwZC05NjZiYTQ5MzE1ZjMiLCJleHAiOjE2OTM0NDMzMzEsImZoaXJVc2VyIjoiaHR0cHM6Ly92ZW5kb3JzZXJ2aWNlcy5lcGljLmNvbS9pbnRlcmNvbm5lY3QtYW1jdXJwcmQtb2F1dGgvYXBpL0ZISVIvUjQvUHJhY3RpdGlvbmVyL2VKaVBjTTZLaXE0bVRrQUZqUDVqeVZRMyIsImlhdCI6MTY5MzQ0MzAzMSwiaXNzIjoiaHR0cHM6Ly92ZW5kb3JzZXJ2aWNlcy5lcGljLmNvbS9pbnRlcmNvbm5lY3QtYW1jdXJwcmQtb2F1dGgvb2F1dGgyIiwic3ViIjoiZUppUGNNNktpcTRtVGtBRmpQNWp5VlEzIn0.dY8zY-9cpH2hEM0toHUPrYajUNgXLcyB1NEAi1raNRuQlP-ymkYIGrnghsreoeMwkVNKtnFxcB-0D3SpkDc5ChKSZ5250YoeRbnXTS6tb6ACVPzQeMGr0dztqFEkRPvQFFaOZCTxTsQv8HbmJlLMZqIU5mzpTxn3Utvlud6aKNk3Ek5K6XG3KBU2Smo2OdoQjjBiFtFaJCPlJ_QVhPpZR6oslv-qtEZEL-rp0NOXWY4fgbX8u5P0oc3vpDR7TMQs6cNAS42A_6QxU6k6QlVZjES1EHJ_sY8Kqux9QsQsnGrFYWrWGzxG6ndGpooN2n5XhaoQEI4DZ1hxbb_4qsMUyQ",
  "location": "e4W4rmGe9QzuGm2Dy4NBqVc0KDe6yGld6HW95UuN-Qd03",
  "loginDepartment": "e4W4rmGe9QzuGm2Dy4NBqVc0KDe6yGld6HW95UuN-Qd03",
  "need_patient_banner": "false",
  "patient": "eJzlzKe3KPzAV5TtkxmNivQ3",
  "smart_style_url": "https://vendorservices.epic.com/interconnect-amcurprd-oauth/api/epic/2016/EDI/HTTP/style/33500004786/I0YwRjJGNHwjQzEyMTI3fCMxQUFCRkZ8I0QzRDhERXwjODZCNTQwfCMwMDAwMDB8MHB4fDEwcHh8fEFyaWFsLCBzYW5zLXNlcmlmfCdTZWdvZSBVSScsIEFyaWFsLCBzYW5zLXNlcmlmfHw%3D.json"
}
```

**Comment by mcjustin** [comment]
> @mcjustin getting closer here, need to decode, clean and parse the content... BUT, I also think much of this is noise and isn't useful in the logs on every launch->authorize request. Thinking I could strip out at least the following keys (or we can white list the ones we want, or keep 'em all as you see fit):
> ...

@pbugni  one use case here is that UCSD wants us to be able to easily tell them details about who has logged in, and which patient(s) they've accessed. 

From your "exclude" list, I'd be inclined to retain these - while they don't make sense to us, it may be of use to them, or at least show due diligence on our part:
```
location
loginDepartment
```

**Review by mcjustin — APPROVED** [review]

File: confidential_backend/config.py [hunk 1]
```diff
@@ -5,6 +5,7 @@
 import os
 import redis
 
+AUTH_TOKEN_LOG_FILTER = os.getenv("AUTH_TOKEN_LOG_FILTER")
```
Comment (ivan-c) [inline confidential_backend/config.py:8]: Minor nitpick: if we ever want to override by env var, we'll want to pass them as a comma separated list
```suggestion
AUTH_TOKEN_LOG_FILTER = os.getenv("AUTH_TOKEN_LOG_FILTER").split(",") if "AUTH_TOKEN_LOG_FILTER" in os.env else None
```

**Review by ivan-c — APPROVED** [review]

**Comment by mcjustin** [comment]
@pbugni  I just retested this; the url was
```
https://confidentialbackend.embedhw.dev.cirg.uw.edu/auth/launch?iss=https%3A%2F%2Fvendorservices.epic.com%2Finterconnect-amcurprd-oauth%2Fapi%2FFHIR%2FR4&launch=eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJ1cm46ZXBpYzphcHBvcmNoYXJkLmN1cnByb2QiLCJjbGllbnRfaWQiOiJjOTE2ODg5Zi00ZTMzLTRkZmEtOTgwZC05NjZiYTQ5MzE1ZjMiLCJlcGljLmVjaSI6InVybjplcGljOkN1cnJlbnQtQXBwLU9yY2hhcmQtUHJvZHVjdGlvbiIsImVwaWMubWV0YWRhdGEiOiJWSUEtdjN3cUc0S2NTSzBoUVN3RDZGcThZdGhERkpIZE11cGFLSVM5ZDJqZnF0VVFIaHFHdVhraDdBdnlfUGlnNlNUSmtTcXBhdHh6cndJXzdJRXhZS0lwSWZwZlNFX1hZeTN6eHNoczF3ZEVOa01ra25FWDBJQUM1ODdWZFJreiIsImVwaWMudG9rZW50eXBlIjoibGF1bmNoIiwiZXhwIjoxNjkzOTU4NTgyLCJpYXQiOjE2OTM5NTgyODIsImlzcyI6InVybjplcGljOmFwcG9yY2hhcmQuY3VycHJvZCIsImp0aSI6ImRmM2Q1MWY5LTIwMWEtNGM5NS05OGNkLTBjNmM4M2M3YjU5YiIsIm5iZiI6MTY5Mzk1ODI4Miwic3ViIjoiZUppUGNNNktpcTRtVGtBRmpQNWp5VlEzIn0.m5fEvnakOyE6W0hJ0Sc8M9kMPLRI9qupNtxmwo70qfE19O5s7u09gDIif0cPZ8no-BoTqvUf316FaQNkEC6AC1-GiiJjganQtT7Dsksl6l0Wh8j5pZ6k83WsTRIXraPZMtMrCh2TJcRzblI2lojiRi_6uSb2HUnaYhoauMvHjR1A8ZTRTBMMVFCDwOxYUYo_ohcXtpL6xUCYnh9jPyK8UuujnwWeiuBbgYFSq154S6veAVncL607nsw9cavLQVDUJROz_jYUGzyrlMEUXMCiEbF8yhsNpCiMqVWZ38RZOHVTBDpVUxQ_g4I07LwdlLEuSpk6hZg4XrRkDg5Im0XJvw
```

**Comment by mcjustin** [comment]
@pbugni  not to distract from my message an hour ago, but I've just edited the `SOF_CLIENT_ID` and `SOF_CLIENT_SECRET` at embedhw-dev's confidentialbackend.env. This, to point it at a new app configured at fhir.epic.com using a CIRG-shared gmail acct. 
Not a rush to redeploy - you may want to review the token logging in its current config.

**Comment by pbugni** [comment]
@mcjustin looks reasonable to me - see near bottom of https://logs.embedhw.dev.cirg.uw.edu/events

**Comment by pbugni** [comment]
@mcjustin , FYI, merged this and redeployed out of the main branch on `embedhw.dev.cirg.uw.edu`

**Pull request merged**